### PR TITLE
chore: Type improvements in http api generator

### PR DESCRIPTION
**Pull request by aahung**
### Issue #, if available

TYPES!

### Description of changes

This also fixes an incorrect error message, see `error_httpapi_mtls_configuration_invalid_type` and comments for detail

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/api/http_api_generator.py [hunk 1]
```diff
@@ -285,66 +294,85 @@ def _construct_api_domain(self, http_api, route53_record_set_groups):  # type: i
                     domain.MutualTlsAuthentication["TruststoreVersion"] = mutual_tls_auth["TruststoreVersion"]  # type: ignore[attr-defined]
             else:
                 raise InvalidResourceException(
-                    mutual_tls_auth,
+                    self.logical_id,
```
Comment (aahung) [inline samtranslator/model/api/http_api_generator.py:297]: Here is the fix of the incorrect error message

File: tests/translator/output/error_httpapi_mtls_configuration_invalid_type.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
-  "errorMessage": "Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [['TruststoreUri', 'TruststoreVersion']] is invalid. MutualTlsAuthentication must be a map with at least one of the following fields ['TruststoreUri', 'TruststoreVersion'].",
+  "errorMessage": "Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [MyApi] is invalid. MutualTlsAuthentication must be a map with at least one of the following fields ['TruststoreUri', 'TruststoreVersion'].",
   "errors": [
     {
-      "errorMessage": "Resource with id [['TruststoreUri', 'TruststoreVersion']] is invalid. MutualTlsAuthentication must contains one of the following fields ['TruststoreUri', 'TruststoreVersion']."
+      "errorMessage": "Resource with id [MyApi] is invalid. MutualTlsAuthentication must contains one of the following fields ['TruststoreUri', 'TruststoreVersion']."
```
Comment (hoffa) [inline tests/translator/output/error_httpapi_mtls_configuration_invalid_type.json:5]: Why? I was hoping this would be a refactor.

Comment (aahung) [reply tests/translator/output/error_httpapi_mtls_configuration_invalid_type.json:5]: See my comment on line http_api_generator.py297

**Review by hoffa — APPROVED** [review]
🚢

**Review by GavinZZ — APPROVED** [review]

**Review by xazhao — APPROVED** [review]

**Pull request merged**